### PR TITLE
Add alpha layer support to gray2rgb

### PR DESCRIPTION
**Pull request by stefanv**

### CODE REVIEW

File: skimage/color/tests/test_colorconv.py [hunk 1]
```diff
@@ -454,6 +454,23 @@ def test_gray2rgb_rgb():
     assert_equal(x, y)
 
 
+def test_gray2rgb_alpha():
+    x = np.random.random((5, 5, 4))
+    assert_equal(gray2rgb(x, alpha=None).shape, (5, 5, 4))
```
Comment (blink1073) [inline skimage/color/tests/test_colorconv.py]: This is an API change, no?

Comment (stefanv) [inline skimage/color/tests/test_colorconv.py]: No, I believe this is how it worked before.

Comment (blink1073) [inline skimage/color/tests/test_colorconv.py]:
```python
In [10]: from skimage.color import rgb2gray

In [11]: import numpy as np

In [12]: x = np.random.random((5, 5, 4))

In [13]: rgb2gray(x).shape
Out[13]: (5, 5)
```

Comment (blink1073) [inline skimage/color/tests/test_colorconv.py]: Whoops, brain fart, I was thinking the other way around.

```python
In [16]: gray2rgb(x).shape
Out[16]: (5, 5, 4)
```

**Comment by JDWarner** [comment]
Seems this might be useful in the reverse kwarg also.

**Comment by stefanv** [comment]
Can you expand?

**Comment by JDWarner** [comment]
I worded my comment poorly - what I meant to say was that alpha channels can be used/applied in almost any image color formulation. This kwarg would be of assistance for any `*2rgb` function, and a grayscale image with shape `(m, n, 4)` with R == G == B and an alpha layer would also be of use, e.g., I used this latter method for the panorama tutorial.

Mainly was wondering if this logic should be contained in `gray2rgb` or factored out and applied a bit more broadly.

**Comment by stefanv** [comment]
Thanks for explaining, Josh--I agree that a utility function would 
be helpful here.

File: skimage/color/colorconv.py [hunk 2]
```diff
@@ -734,11 +737,39 @@ def gray2rgb(image):
...
         image = image[..., np.newaxis]
-        return np.concatenate(3 * (image,), axis=-1)
+
+        if alpha:
+            alpha_layer = (np.ones_like(image) * dtype_limits(image)[1])
+            return np.concatenate(3 * (image,) + (alpha_layer,), axis=-1)
+        else:
+            return np.concatenate(3 * (image,), axis=-1)
+
+        return image
```
Comment (pratapvardhan) [inline skimage/color/colorconv.py]: @stefanv -- it would never return this case I suspect?

Comment (stefanv) [inline skimage/color/colorconv.py]: Yes, you're right, that line can go.

Comment (stefanv) [inline skimage/color/colorconv.py]: Please test it and make a PR, since this has been merged.

**Pull request merged**